src/main/kotlin/org/rust/ide/inspections/UnnecessaryParenthesisInspection.kt

```diff
@@ -0,0 +1,35 @@
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.LocalQuickFixBase
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElementVisitor
+import org.rust.lang.core.psi.RustIfExpr
+import org.rust.lang.core.psi.RustVisitor
+import org.rust.lang.core.psi.impl.RustParenExprImpl
+
+/**
+ * Created by zjh on 16/3/8.
+ */
+class UnnecessaryParenthesisInspection : RustLocalInspectionTool() {
+    override fun getDisplayName() = "Unnecessary parenthesis"
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor =
+        object : RustVisitor() {
+            override fun visitIfExpr(expr: RustIfExpr) {
+                val exprText = expr.expr?.text
+                if (exprText?.startsWith("(")!! && exprText?.endsWith(")")!!) {
+                    holder.registerProblem(expr.expr!!, "Unnecessary parenthesis", RemoveParenthesis);
+                }
+            }
+        }
+
+    object RemoveParenthesis : LocalQuickFixBase("Remove parenthesis") {
+        override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
+            val expr = descriptor.psiElement as RustParenExprImpl
```

Comment:
I think you should do something like 

```Rust
val ifExpr = descriptor.psiElement
val cond = ifExpr.expr
cond.replace(cond.expr)
```

However, `RustPatensExpr` lacks `expr` accessor for some reason :( It must be a bug in our grammar, so I don't think it is possible to implement this quick fix just yet. 

So let's drop quick fix and left only the inspection itself for now